static/js/components/forms/RegisterDetailsForm.js

```diff
@@ -31,35 +35,47 @@ const INITIAL_VALUES = {
   }
 }
 
-const RegisterDetailsForm = ({ onSubmit, countries }: Props) => (
-  <Formik
-    onSubmit={onSubmit}
-    validationSchema={legalAddressValidation.concat(passwordValidation)}
-    initialValues={INITIAL_VALUES}
-    render={({ isSubmitting, setFieldValue, setFieldTouched, values }) => (
-      <Form>
-        <LegalAddressFields
-          countries={countries}
-          setFieldValue={setFieldValue}
-          setFieldTouched={setFieldTouched}
-          values={values}
-          includePassword={true}
-        />
-        <div>
-          <div className="required">*=Required</div>
-          <div className="row submit-row no-gutters justify-content-end">
-            <button
-              type="submit"
-              className="btn btn-outline-danger large-font"
-              disabled={isSubmitting}
-            >
-              Continue
-            </button>
+const RegisterDetailsForm = ({
+  onSubmit,
+  countries,
+  includePassword
+}: Props) => {
+  if (!includePassword && INITIAL_VALUES.password) {
+    // $FlowFixMe
+    delete INITIAL_VALUES.password
```

Comment:
This shouldn't be mutating a `const`, it's probably better to have two consts, one without `password, the other:

```js
const INITIAL_WITH_PASSWORD = {...INITITAL_WITHOUT_PASSWORD, password: ""}
```